src/gtc/oir.py

```diff
@@ -120,13 +120,18 @@ class ScalarDecl(Decl):
     pass
 
 
+class LocalScalar(ScalarDecl):
```

Comment:
Is it needed, se also my other suggestions below

Reply:
Can not see any other suggestions below…

Reply:
Below in the file, i.e. the previous comment about fixing params. I meant, is there a good reason to distinguish ScalarDecl as parameter and LocalScalar as a declaration inside the body.

Reply:
I guess so. First, they are declared at a different point, ScalarDecl are global parameters, passed as stencil arguments, LocalScalars are local to a horizontal execution. Also, they mean very different things, the ScalarDecl stands for one scalar for the whole domain, LocalScalar stands for a different scalar at each grid point. Further, the lowering to GTCPP is simpler (and maybe other possible IRs) as the distinction has to be handled there, too.

Reply:
But then it shouldn't inherit from ScalarDecl?

Reply:
True, fixed!